apps/c/redis/redis.patch

```diff
@@ -0,0 +1,236 @@
+         serverPanic("Unrecoverable error creating TCP socket accept handler.");
+     }
++    printf("initServer: createSocketAcceptHandler1 pass\n");
+     if (createSocketAcceptHandler(&server.tlsfd, acceptTLSHandler) != C_OK) {
+         serverPanic("Unrecoverable error creating TLS socket accept handler.");
+     }
++    printf("initServer: createSocketAcceptHandler2 pass\n");
+     if (server.sofd > 0 && aeCreateFileEvent(server.el,server.sofd,AE_READABLE,
+         acceptUnixHandler,NULL) == AE_ERR) serverPanic("Unrecoverable error creating server.sofd file event.");
+
+-
++    printf("initServer: aeCreateFileEvent1 pass\n");
+     /* Register a readable event for the pipe used to awake the event loop
+      * from module threads. */
+     if (aeCreateFileEvent(server.el, server.module_pipe[0], AE_READABLE,
+@@ -2651,11 +2656,14 @@ void initServer(void) {
+             serverPanic(
+                 "Error registering the readable event for the module pipe.");
+     }
++    printf("initServer: aeCreateFileEvent2 pass\n");
+
+     /* Register before and after sleep handlers (note this needs to be done
+      * before loading persistence since it is used by processEventsWhileBlocked. */
+     aeSetBeforeSleepProc(server.el,beforeSleep);
++    printf("initServer: aeSetBeforeSleepProc pass\n");
+     aeSetAfterSleepProc(server.el,afterSleep);
++    printf("initServer: aeSetBeforeSleepProc pass\n");
+
+     /* 32 bit instances are limited to 4GB of address space, so if there is
+      * no explicit limit in the user provided configuration we set a limit
+@@ -2668,15 +2676,22 @@ void initServer(void) {
+     }
+
+     if (server.cluster_enabled) clusterInit();
++    printf("initServer: clusterInit pass\n");
+     scriptingInit(1);
++    printf("initServer: scriptingInit pass\n");
+     functionsInit();
++    printf("initServer: functionsInit pass\n");
+     slowlogInit();
++    printf("initServer: slowlogInit pass\n");
+     latencyMonitorInit();
++    printf("initServer: latencyMonitorInit pass\n");
+
+     /* Initialize ACL default password if it exists */
+     ACLUpdateDefaultUserPassword(server.requirepass);
++    printf("initServer: ACLUpdateDefaultUserPassword pass\n");
+
+     applyWatchdogPeriod();
++    printf("initServer: applyWatchdogPeriod pass\n");
+
+     if (server.maxmemory_clients != 0)
+         initServerClientMemUsageBuckets();
+@@ -4093,6 +4108,9 @@ int prepareForShutdown(int flags) {
+         flags = (flags & ~SHUTDOWN_SAVE) | SHUTDOWN_NOSAVE;
+
+     server.shutdown_flags = flags;
++    char buf[128];
++    sprintf(buf, "shutdown flag = %d", flags);
++    puts(buf);
+
+     serverLog(LL_WARNING,"User requested shutdown...");
+     if (server.supervised_mode == SUPERVISED_SYSTEMD)
+@@ -6108,7 +6126,7 @@ int checkIgnoreWarning(const char *warning) {
+             break;
+     }
+     sdsfreesplitres(argv,argc);
+-    return j < argc;
++    return j <= argc;
```

Comment:
Why use `<=`?

Reply:
This is a way to fake system check on aarch64. Because Redis will do some checks, using fork/mmap, which are not implemented and will cause `exit()`, in server.c line 7133-7145, there should be somewhere to bypass this check. And this is it.